js/components/graph/Node.js

```diff
@@ -2,57 +2,99 @@ import PropTypes from "prop-types";
 import React from "react";
 import { refLookUp } from "../common/utils";
 
+/** React component class representing a Node on the graph */
 export default class Node extends React.Component {
+  /** Create a node */
   constructor(props) {
     super(props);
     var state = localStorage.getItem(this.props.JSON.id_);
+
+    /**
+     * Unselected states:
+     *  - takeable: all prerequisites are satisfied (or no prereqs/parents)
+     *  - inactive: missing some prerequisites
+     * 
+     * Selected states:
+     *  - active: all prerequisites are satisfied
+     *  - overridden: missing some prerequisites (will have a red border)
+     */
+
     if (this.props.editMode) {
       state = ""; // TODO: define what editMode is
     } else if (state === null) {
+      // sets state to takeable if it has no parent nodes (prereqs)
       state = this.props.parents.length === 0 ? "takeable" : "inactive";
     }
+
     this.state = {
       status: state,
-      selected: ["active", "overridden"].indexOf(state) >= 0
+      selected: ["active", "overridden"].indexOf(state) >= 0 // checks if state means its selected
```

Comment:
Remove this comment. Instead, it's better to add a note in the class documentation above that `selected` is a state.